errors/error_test.go

```diff
@@ -0,0 +1,217 @@
+// Copyright 2022 Mineiros GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package errors_test
+
+import (
+	stderrors "errors"
+	stdfmt "fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/madlambda/spells/assert"
+	"github.com/mineiros-io/terramate/errors"
+)
+
+var E = errors.E
+
+var (
+	syntaxError   errors.Kind = "syntax error"
+	tmSchemaError errors.Kind = "terramate schema error"
+	tfSchemaError errors.Kind = "terraform schema error"
+)
+
+func TestNoArgs(t *testing.T) {
+	defer func() {
+		err := recover()
+		if err == nil {
+			t.Fatal("E() did not panic")
+		}
+	}()
+	_ = E()
```

Comment:
Maybe we could have something like a assert.Panic(func() {}) that executes the given function and checks it panics.

Reply:
cool